Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+    DoubleWrite = 0;
+    ReadRace = 1;
+    UndeclaredOrderedRead = 2;
+    MissingSourceDependency = 3;
+    UndeclaredReadCycle = 4;
+    UndeclaredOutput = 5;
+    ReadUndeclaredOutput = 6;
+    WriteInSourceSealDirectory = 7;
+    WriteInUndeclaredSourceRead = 8;
+    WriteInExistingFile = 9;
+    WriteOnAbsentPathProbe = 10;
+    AbsentPathProbeUnderUndeclaredOpaque = 11;
+    WriteToTempPathInsideSharedOpaque = 12;
+}
+
+enum FileMonitoringViolationAnalyzer_AccessLevel_XLGpp{
+    FileMonitoringViolationAnalyzer_AccessLevel_Read = 0;
+    FileMonitoringViolationAnalyzer_AccessLevel_Write = 1;
+}
+
+enum PipCacheMissType_XLGpp{
+    Invalid = 0;
+    MissForDescriptorsDueToStrongFingerprints = 53;
+    MissForDescriptorsDueToWeakFingerprints = 54;
+    MissForDescriptorsDueToArtificialMissOptions = 55;
+    MissForCacheEntry = 56;
+    MissDueToInvalidDescriptors = 57;
+    MissForProcessMetadata = 58;
+    MissForProcessMetadataFromHistoricMetadata = 59;
+    MissForProcessOutputContent = 60;
+    Hit = 51;
+    MissForProcessConfiguredUncacheable = 61;
+}
+
+enum ExecutionSampler_LimitingResource_XLGpp{
+    GraphShape = 0;
+    ExecutionSampler_LimitingResource_CPU = 1;
+    Disk = 2;
+    Memory = 3;
+    ConcurrencyLimit = 4;
+    ProjectedMemory = 5;
+    Semaphore = 6;
+    Other = 7;
+}
+
+enum CreationDisposition_XLGpp{
+    // First enum value in proto must be 0
+    CreationDisposition_DEFAULT_ZERO = 0;
+    CREATE_NEW = 1;
+    CREATE_ALWAYS = 2;
+    OPEN_EXISTING = 3;
+    OPEN_ALWAYS = 4;
+    TRUNCATE_EXISTING = 5;
+}
+
+enum DesiredAccess_XLGpp{
+    // First enum value in proto must be 0
+    DesiredAccess_DEFAULT_ZERO = 0;
+    DELETE = 0x00010000;
+    SYNCHRONIZE = 0x00100000;
+    FILE_APPEND_DATA = 0x00000004;
+    FILE_WRITE_EA = 0x00000010;
+    FILE_WRITE_ATTRIBUTES = 0x00000100;
+    FILE_WRITE_DATA = 0x00000002;
+    GENERIC_ALL = 0x10000000;
+    GENERIC_EXECUTE = 0x20000000;
+    GENERIC_WRITE = 0x40000000;
+    GENERIC_READ = -2147483648;
+}
+
+enum FlagsAndAttributes_XLGpp{
+    // First enum value in proto must be 0
+    FlagsAndAttributes_DEFAULT_ZERO = 0;
+    FILE_ATTRIBUTE_READONLY = 0x00000001;
+    FILE_ATTRIBUTE_HIDDEN = 0x00000002;
+    FILE_ATTRIBUTE_SYSTEM = 0x00000004;
+    FILE_ATTRIBUTE_ARCHIVE = 0x00000020;
+    FILE_ATTRIBUTE_NORMAL = 0x00000080;
+    FILE_ATTRIBUTE_TEMPORARY = 0x00000100;
+    FILE_ATTRIBUTE_OFFLINE = 0x00001000;
+    FILE_ATTRIBUTE_ENCRYPED = 0x00004000;
+    FILE_FLAG_OPEN_NO_RECALL = 0x00100000;
+    FILE_FLAG_OPEN_REPARSE_POINT = 0x00200000;
+    FILE_FLAG_SESSION_AWARE = 0x00800000;
+    FILE_FLAG_POSIX_SEMANTICS = 0x01000000;
+    FILE_FLAG_BACKUP_SEMANTICS = 0x02000000;
+    FILE_FLAG_DELETE_ON_CLOSE = 0x04000000;
+    FILE_FLAG_SEQUENTIAL_SCAN = 0x08000000;
+    FILE_FLAG_RANDOM_ACCESS = 0x10000000;
+    FILE_FLAG_NO_BUFFERING = 0x20000000;
+    FILE_FLAG_OVERLAPPED = 0x40000000;
+    FILE_FLAG_WRITE_THROUGH = -2147483648;

```

Comment:
>-2147483648 [](start = 30, length = 11)

it loos like some 'magic number' - why are you not using 0x80000000?

Reply:
0x80000000 is out of bounds for enums. But 0x80000000 == -2147483648 from a signed perspective, and protobuf accepts the negative number

---
In reply to: [308893360](https://github.com/microsoft/BuildXL/pull/671#discussion_r308893360) [](ancestors = 308893360)